### PR TITLE
Test examples/ping-*/README.md

### PR DESCRIPTION
**Pull request by Motiejus**
Using [cram][1] to execute and verify instructions in the following files:

* `examples/ping-json/README.md`
* `examples/ping-thrift-gen/README.md`
* `examples/ping-thrift/README.md`

Currently, all of the tests are failing due to incomplete instructions in the README, but opening the pull  request anyway to set the path forward to fixing them.

[1]: https://pypi.python.org/pypi/cram

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,12 @@ go:
   - 1.6
   - 1.5
 
+sudo: false
```
Comment (dansimau) [inline .travis.yml]: Why is `sudo: false` required? IIRC, this is the default

Comment (Motiejus) [inline .travis.yml]: Leftovers from development process. Good catch, removed.

**Comment by dansimau** [comment]
LGTM, happy to merge, though I do have this philosophical thought:

Is storing the tests in the README actually a good idea? I know it seems cool because it's less files or whatever, but it is sane/obvious/intuitive?

Note we have sacrificed formatting in the README to do this:
https://github.com/uber/ringpop-go/blob/32aed6541ce6ae53100d81f82a47ce7fe346e90a/examples/ping-json/README.md#running-the-example

I don't feel strongly enough about this to block the merge of this PR. Practically, I don't think this will cause us a huge issue/effort in the future. But I do think tests in the README is a bit strange and may increase the average WTFs/min in the codebase.

**Comment by Motiejus** [comment]
@dansimau thanks for the comment, and this is indeed a good philosophical discussion.

If we look at what we want to achieve:

* Have clear instructions on how to build and use the project.
* Make sure the instructions are sane (= automatically tested).

I see two approaches:

1. *More common:* separate the documentation generation and testing. Usually deliverable is a docbook- or a sphinx-like documentation artifact, which, in the back, nobody cares how it's generated.
2. *Less common:* test the actual final documentation.
3. Move the steps from README to a script, and test the script.

(1) is a standard practice. However, since github does not really allow it easily (I want the README to be readable as-is).
(2) gives more wtfs/minute, but it will work, as long as people keep adding the correct examples.
(3) seems to be least wtfs/minute, but leads to a least readable README.

**Comment by Motiejus** [comment]
cc @severb can you check out the examples?

File: test/run-example-tests [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+#
+# Runs the ringpop-admin tests.
+#
+# This script launches a ringpop cluster using tick-cluster and then runs
+# a series of tests against the ringpop-admin command output.
+#
+set -euo pipefail
+
+
+declare cram_opts="-v --shell=/bin/bash"
```
Comment (thanodnl) [inline test/run-example-tests]: I don't like the formatting of the README.md files now. Due to only two spaces they are hard to read on github. if we add `--indent=4` to the cram options we can indent the code blocks by 4 spaces which will put a box around them in markdown making it a bit easier to read IMHO.

**Comment by thanodnl** [comment]
LGTM

**Comment by Motiejus** [comment]
@dansimau we actually fixed the formatting in README. Can you re-check?

**Comment by dansimau** [comment]
LGTM

**Pull request merged**